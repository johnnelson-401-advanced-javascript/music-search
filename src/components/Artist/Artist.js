import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function Artist({ name, id, disamb }) {
  return (
    <Link to={`/artist/${id}/${name}`}>
      <h2>{name}{disamb ? `: ${disamb}` : null}</h2>
    </Link>
  );
}
Artist.propTypes = {
  name: PropTypes.string.isRequired,
  disamb: PropTypes.string,
  id: PropTypes.string.isRequired
};
