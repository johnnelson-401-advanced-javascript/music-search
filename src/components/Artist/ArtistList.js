import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

export default function ArtistList({ artists, handlePageBackward, handlePageForward }) {

  const artistElements = artists.map(artist => (
    <li key={artist.id}>
      <Artist name={artist.name} id={artist.id} disamb={artist.disamb} />
    </li>
  ));

  return (
    <>
      <button onClick={handlePageBackward}>Back</button>
      <button onClick={handlePageForward}>Next</button>
      <ul>
        {artistElements}
      </ul>
    </>
  );
}
ArtistList.propTypes = {
  handlePageBackward: PropTypes.func.isRequired,
  handlePageForward: PropTypes.func.isRequired,
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    disamb: PropTypes.string,
  })).isRequired
};
