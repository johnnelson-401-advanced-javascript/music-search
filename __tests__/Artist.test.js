import React from 'react';
import { shallow } from 'enzyme';
import Artist from '../src/components/Artist/Artist';

describe('Artist component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist name='John' id='2' disamb='' />);
    expect(wrapper).toMatchSnapshot();
  });
});
