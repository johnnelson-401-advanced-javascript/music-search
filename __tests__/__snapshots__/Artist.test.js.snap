// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<Link
  to="/artist/2/John"
>
  <h2>
    John
  </h2>
</Link>
`;
